AudioLooper/LooperThreadC.h

```diff
@@ -53,19 +62,30 @@ class LooperThreadC: public LooperC
 			setTrack(getAudioFile());
 			break;
 		case APPLICATION_FUNCTIONS::PLAY:
-			playTrack();
+			if (!(m_recorder->isRecording())) {
+				playTrack();
+			}
 			break;
 		case APPLICATION_FUNCTIONS::STOP:
-			stopTrack();
+			if (m_recorder->isRecording())
+			{
+				m_recorder->Stop();
+			}
+			else if (isPlaying())
+			{
+				stopTrack();
+			}
 			break;
 		case APPLICATION_FUNCTIONS::RECORD_TO_FILE:
-			m_recorder->Record();
-			break;
-		case APPLICATION_FUNCTIONS::RECORD_FROM_FILE:
-
+			if (getAudioFile() != "NONE") {
```

Comment:
When I try to record for the first time, this conditional is preventing the Record() from kicking off.
Would:
`getAudioFile() == "NONE"`
be a solution?

Reply:
One consideration: maybe get rid of the conditional all together? If the user clicks record again and there is already an preexisting audio file, perhaps just delete the old one and replace it with the new one?